server/src/test/java/org/opensearch/search/aggregations/startree/MetricAggregatorTests.java

```diff
@@ -555,26 +554,36 @@ public QueryBuilder getTermsQueryBuilder() {
         }
 
         public QueryBuilder getRangeQueryBuilder() {
-            return new RangeQueryBuilder(fieldName).from(valueSupplier.get())
+            RangeQueryBuilder qb = new RangeQueryBuilder(fieldName).from(valueSupplier.get())
                 .to(valueSupplier.get())
                 .includeLower(randomBoolean())
                 .includeUpper(randomBoolean());
+            return qb;
         }
 
         public QueryBuilder getBoolQueryBuilder() {
             // MUST only
-            BoolQueryBuilder mustOnly = new BoolQueryBuilder().must(getTermQueryBuilder()).must(getRangeQueryBuilder());
+            BoolQueryBuilder mustOnly = new BoolQueryBuilder().must(getTermQueryBuilder());
+            // Keyword terms are not always supported for range queries.
+            if (fieldType.equals(DimensionTypes.KEYWORD.name().toLowerCase(Locale.ROOT))) {
+                mustOnly.must(getTermQueryBuilder());
+            } else {
+                mustOnly.must(getRangeQueryBuilder());
+            }
 
             // MUST with nested SHOULD on same dimension
             BoolQueryBuilder mustWithShould = new BoolQueryBuilder().must(getTermQueryBuilder())
-                .must(
-                    new BoolQueryBuilder().should(new TermQueryBuilder(fieldName, valueSupplier.get()))
-                        .should(new TermQueryBuilder(fieldName, valueSupplier.get()))
-                );
+                .must(new BoolQueryBuilder().should(getTermQueryBuilder()).should(getTermQueryBuilder()));
 
             // SHOULD only on same dimension
-            BoolQueryBuilder shouldOnly = new BoolQueryBuilder().should(new TermQueryBuilder(fieldName, valueSupplier.get()))
-                .should(new RangeQueryBuilder(fieldName).from(valueSupplier.get()).to(valueSupplier.get()));
+            BoolQueryBuilder shouldOnly = new BoolQueryBuilder().should(getTermQueryBuilder());
+
+            // Keyword terms are not always supported for range queries.
```

Comment:
nit: // Keyword terms are not ~always~ supported for range queries.

(remove always)